tests/test_io.py

```diff
@@ -192,18 +192,39 @@ def test_file_extension(path, extension):
 
 
 @pytest.mark.parametrize(
-    'dir_list,recursive',
+    'dir_list,expected,recursive,hidden',
     [
-        ([], False),
-        ([], True),
-        (['a', 'b', 'c'], False),
-        (['a', 'b', 'c'], True),
-        (['a'], False),
-        (['a'], True),
-        (['a', os.path.join('a', 'b'), os.path.join('a', 'b', 'c')], True),
+        ([], [], False, False),
+        ([], [], True, False),
+        (['a', 'b', 'c'], ['a', 'b', 'c'], False, False),
+        (['a', 'b', 'c'], ['a', 'b', 'c'], True, False),
+        (['a'], ['a'], False, False),
+        (['a'], ['a'], True, False),
+        (
+            ['a', os.path.join('a', 'b'), os.path.join('a', 'b', 'c')],
+            ['a', os.path.join('a', 'b'), os.path.join('a', 'b', 'c')],
+            True,
+            False,
+        ),
+        # hidden
+        (['a', '.b'], ['a'], True, False),
+        (['a', '.b'], ['.b', 'a'], True, True),
```

Comment:
I guess this means hidden folders are always placed at the beginning in the sorted returned path list, correct?

Reply:
Yes, it's because we sort the files at the end. But maybe we should first sort the non-hidden files and then append the hidden files so that instead of:

  ```
.file.wav
file.wav
```

we return:

```
file.wav
.file.wav
```

The other question is, it makes sense to only sort within sub-directories, i.e. instead of:

```
a/file.wav
file.wav
```

we return:

```
file.wav
a/file.wav
```